keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -438,7 +430,7 @@ def train_step(self, data):
         self.optimizer.apply_gradients(zip(gradients, trainable_vars))
 
         # Early exit for no train time metrics
-        if not self.evaluate_train_time_metrics:
+        if len(super().metrics) == 0:
```

Comment:
With this change, RetinaNet fails again on TPU (error: XLA cannot compile NMS). It worked on the previous commit with the evaluate_train_time_metrics=False. I do not have any metrics in .compile().

Reply:
I think the issue is that super().metrics always has loss in it, so we might have to store a flag in `compile()` indicating if a metric was passed or not.

Reply:
> With this change, RetinaNet fails again on TPU (error: XLA cannot compile NMS). It worked on the previous commit with the evaluate_train_time_metrics=False. I do not have any metrics in .compile().

It is always a surprise what things compile and what not in the library. 
I hope that soon or later we could regularly add compiler decorated versions to our tests (eventually with expected failures).

As our NMS layer it is using `tf.image.combined_non_max_suppression` it is supported only enabling the new MLIR XLA bridge:
https://github.com/keras-team/keras-cv/pull/1021#issuecomment-1317925675

As XLA [it is still fail fast](https://github.com/tensorflow/tensorflow/issues/57923) the next issue is:

```python
INVALID_ARGUMENT: TF to XLA legalization ... 'tf.TensorListSetItem' op found tf.TensorListSetItemOp on unknown TensorList.
```

That i suppose it is caused by our `TensorArray` manipulation:
https://github.com/keras-team/keras-cv/blob/51260f0689df27525ee307dd15caca87800f0765/keras_cv/layers/object_detection/non_max_suppression.py#L144-L171


Do you have a more atomic test/input for this?
```python
@tf.function(jit_compile=True)
def _decode_nms_boxes_to_tensor(self, nmsed_boxes):
```

So that we could open a ticket in TF with a minimized but standalone gist.